lib/Extension/LanguageServerCodeTransform/LspCommand/GenerateDecoratorCommand.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerCodeTransform\LspCommand;
+
+use Amp\Promise;
+use Amp\Success;
+use Phpactor\CodeTransform\Domain\Exception\TransformException;
+use Phpactor\Extension\LanguageServerBridge\Converter\TextEditConverter;
+use Phpactor\LanguageServerProtocol\Command;
+use Phpactor\LanguageServerProtocol\WorkspaceEdit;
+use Phpactor\LanguageServer\Core\Server\ClientApi;
+use Phpactor\CodeTransform\Domain\SourceCode;
+
+class GenerateDecoratorCommand extends Command
+{
+    public const NAME = 'generate_decorator';
+
+    private ClientApi $clientApi;
+
+    public function __construct(ClientApi $clientApi)
+    {
+        $this->clientApi = $clientApi;
+    }
+
+    /**
+     * @return Promise<ApplyWorkspaceEditResponse>
+     */
+    public function __invoke(string $uri): Promise
+    {
+        $textDocument = $this->workspace->get($uri);
+
+        try {
+            $textEdits = $this->decorateClass->transform(
```

Comment:
the decorate class is **not** a transformer (in the Phpactor sense).

it should accept the class name and interface name in addition to the text document.

Reply:
Yeah, I have replaced this part with the implementation of the command itself.